bob/src/cluster/quorum.rs

```diff
@@ -239,6 +245,64 @@ impl Quorum {
             Ok(())
         }
     }
+
+    pub(crate) async fn delete_aliens(
+        &self,
+        mut failed_nodes: Vec<String>,
+        key: BobKey,
+    ) -> Result<(), Error> {
+        if failed_nodes.is_empty() {
+            warn!(
+                "DELETE[{}] trying to aliens, but there are no failed nodes: unreachable",
+                key
+            );
+            return Err(Error::internal());
+        }
+        trace!("selection of free nodes available for data writing");
+        let (primary, secondary) = self.mapper.get_support_nodes(key, failed_nodes.len());
+        debug!(
+            "DELETE[{}] sup delete nodes, primary: {:?} secondary: {:?}",
+            key, &primary, &secondary
+        );
+        let primary_need_remote_backup: Vec<_> = failed_nodes.drain(..primary.len()).collect();
+        let secondary_need_remote_backup: Vec<_> = failed_nodes.drain(..secondary.len()).collect();
+        let queries: Vec<_> = primary
+            .into_iter()
+            .zip(primary_need_remote_backup)
+            .map(|(node, remote_node)| (node, DeleteOptions::new_force_alien(vec![remote_node])))
+            .chain(
+                secondary
+                    .into_iter()
+                    .zip(secondary_need_remote_backup)
+                    .map(|(node, remote_node)| (node, DeleteOptions::new_alien(vec![remote_node]))),
+            )
+            .collect();
+        debug!("DELETE[{}] additional alien requests: {:?}", key, queries);
+        if let Err(sup_nodes_errors) = delete_sup_nodes(key, &queries).await {
+            debug!("support nodes errors: {:?}", sup_nodes_errors);
+            failed_nodes.extend(
```

Comment:
Logic is broken: you should not change `failed_nodes`, because after `delete_sup_nodes` it should contain only originally failed nodes!